e2e/cypress/integration/messaging/autocomplete_spec.js

```diff
@@ -0,0 +1,333 @@
+            cy.get(`#postMessageText_${postId}`).should('contain', `${testUser.username}`);
+        });
+    });
+
+    it('MM-T2201 @ autocomplete - first name', () => {
+        cy.visit(`/${testTeam.name}/channels/town-square`).wait(TIMEOUTS.FIVE_SEC);
+
+        // # Clear then type @  and user first names
+        cy.get('#post_textbox').should('be.visible').clear().type(`@${testUser.first_name}`).then(() => {
+            // * Verify that the item is displayed or not as expected.
+            cy.get('#suggestionList').within(() => {
+                cy.findByText(displayNameTestUser).should('be.visible');
+                cy.findByText(displayNameOtherUser).should('not.be.visible');
+            });
+        });
+
+        // # Post user mention
+        cy.get('#post_textbox').type('{enter}{enter}');
+
+        // # Check that the user name has been posted
+        cy.getLastPostId().then((postId) => {
+            cy.get(`#postMessageText_${postId}`).should('contain', `${testUser.username}`);
+        });
+    });
+
+    it('MM-T2203 @ autocomplete - not email', () => {
+        cy.visit(`/${testTeam.name}/channels/town-square`).wait(TIMEOUTS.FIVE_SEC);
+
+        // # Clear then type @ and user email
+        cy.get('#post_textbox').should('be.visible').clear().type(`@${testUser.email}`);
+
+        // * Verify that the item is displayed or not as expected.
+        cy.get('#suggestionList').should('not.be.visible');
+    });
+
+    it('MM-T2206 @ autocomplete - not in channel (center), have permission to add (public channel)', () => {
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`).wait(TIMEOUTS.FIVE_SEC);
+
+        // # Clear then type @
+        cy.get('#post_textbox').should('be.visible').clear().type('@');
+
+        // * Verify that the suggestion list is visible
+        cy.get('#suggestionList').should('be.visible');
+
+        // # Type user name
+        cy.get('#post_textbox').type(`${notInChannelUser.username}`);
+
+        // # Post user mention
+        cy.get('#post_textbox').type('{enter}{enter}');
+
+        const message = `@${notInChannelUser.username} did not get notified by this mention because they are not in the channel. Would you like to add them to the channel? They will have access to all message history.`;
+
+        cy.getLastPostId().then((postId) => {
+            // * Verify that the correct system message is displayed or not
+            cy.get(`#postMessageText_${postId}`).should('include.text', message);
+
+            // * Click on the link to add the user to the channel
+            cy.get('a.PostBody_addChannelMemberLink').should('be.visible').click().then(() => {
+                // * Verify that the correct system message is displayed or not
+                cy.getLastPostId().then((npostId) => {
+                    cy.get(`#postMessageText_${npostId}`).should('include.text', `@${notInChannelUser.username} added to the channel by you`);
+                });
```

Comment:
Thanks for updating. However, I'm still getting intermittent failing test here, depending on how fast/slow the system message gets posted. I thought `cy.uiWaitUntilMessagePostedIncludes` before `cy.getLastPostId` will help make it more deterministic. But it might only be my setup so this is non-blocking for me.